src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1731,13 +1731,15 @@ public ValidateSetAttribute(Type valuesGeneratorType)
     /// <summary>
     /// Allows dynamically generate set of values for <see cref="ValidateSetAttribute"/>
     /// </summary>
+#nullable enable
     public interface IValidateSetValuesGenerator
     {
         /// <summary>
         /// Gets valid values.
         /// </summary>
         string[] GetValidValues();
     }
+#nullable restore
```

Comment:
Should it be:
```c#
string?[] GetValidValues();
```
We have internal check and throw if the method returns null but I don't see that we check the strings is not null. 
Thoughts?

Reply:
The PR builds and passes tests as-is and this is an internal API.

Reply:
It doesn't appear this API was designed to allow `null` as a "valid value", but I can't know the design intent for certain.

Reply:
/cc @SeeminglyScience  @vexx32

Reply:
In ValidateSetAttribute class we check the element in ValidateElement() method and throw if it is null.

@xtqqczze Please enhance the XML comment for GetValidValues() with the two design intentions:
- return is not null
- elements is not null

Reply:
Generally if PS encounters $null for what is a string it'll convert it to `string.Empty`, so unless folks are constructing a values generator and handing back `[nullstring]::Value` it _shouldn't_ be a concern for most things, I wouldn't think?

Reply:
In C# applications no conversion to `[nullstring]::Value` exists and ValidateElement() will throws on null element.

Reply:
I wouldn't generally expect null elements to be supported anyway tbh, it kind of defeats the purpose of the values generator if it's handing back null values in the first place?

Reply:
Yes, current behavior is right.